### ISSUE TITLE
[MPMD-8] PMD should generate results in formats other than html

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-pmd-plugin#381]
**[patrick OShea](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=patrickoshea)** opened **[MPMD-8](https://issues.apache.org/jira/browse/MPMD-8?redirect=false)** and commented

Currently the maven2 pmd plugin only creates html output.
It should be enhanced to allow the output of pmd data in other formats.
Also It should be possible to run PMD on a projects without adding it as a report.

Linked to MNG-1158

---

**Attachments:**
- [how to call from pom.xml](https://issues.apache.org/jira/secure/attachment/12717255/how+to+call+from+pom.xml) (_1.36 kB_)
- [PmdPlugin.java](https://issues.apache.org/jira/secure/attachment/12717270/PmdPlugin.java) (_14.30 kB_)

**Issue Links:**
- [MPMD-4](https://issues.apache.org/jira/browse/MPMD-4) PMD should use "sensible" default rulesets

1 votes, 2 watchers

**Comment by jira-importer** [comment]
**[patrick OShea](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=patrickoshea)** commented

I have attached the pmd plugin class which will create and output
pmd data without adding it as a report in the following formats:
Text, XML, CSV, Emacs, Html, YAHTML, SummaryHtml, VBHtml, Papari.

**Comment by jira-importer** [comment]
**[Mike Perham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mperham)** commented

You should be able to run the pmd plugin on a project just by running 'mvn pmd:pmd'.  The output will be emitted to your site report directory (e.g. target/site).

**Comment by jira-importer** [comment]
**[skaze](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=skaze)** commented

Mike, I know it's closed, that's why I commented it, that and the fact that I didn't want my work to go to waste.

But as it's now closed and not being reopened (wouldn't want to mess with your stats) I will rebase, create more patches and assign them to a new JIRA(s)

I am interested in promoting consistency in Maven plugin behaviour and configuration, especially across plugins with common functionality/usage scenerios, something that is inherently hard to maintain with multiple contributors. My changes were made to that end.

Re the bit you didn't understand:-

"Additionally I took the external report generation functionality from the execute mojo and added it to the report mojo. This is useful for integrating the output into other tools (CC, QALab) as perviously stated although i made it optional as the real item of interst is the generated site/pmd.html file."

Pmd:execute generates  a PMD report (xml/html/csv/etc). I added this functionality to the pmd:pmd Mojo that generates the site report so that I don't need to process my 1800 Java source files twice (the Azureus bit torrent client), once for the site report and again for PMD report for QALabs/cruisecontrol integration.
